projects/packages/sync/src/modules/class-comments.php

```diff
@@ -515,4 +521,73 @@ public function expand_comment_ids( $args ) {
 			$previous_interval_end,
 		);
 	}
+
+	/**
+	 * Expand the comment IDs to comment objects and meta before being serialized and sent to the server.
+	 *
+	 * @access public
+	 *
+	 * @param array $args The hook parameters.
+	 * @return array The expanded hook parameters.
+	 */
+	public function extract_comments_and_meta( $args ) {
+		list( $filtered_comments, $previous_end ) = $args;
+		return array(
+			$filtered_comments['objects'],
+			$filtered_comments['meta'],
+			$previous_end,
+		);
+	}
+
+	/**
+	 * Given the Module Configuration and Status return the next chunk of items to send.
+	 * This function also expands the posts and metadata and filters them based on the maximum size constraints.
+	 *
+	 * @param array $config This module Full Sync configuration.
+	 * @param array $status This module Full Sync status.
+	 * @param int   $chunk_size Chunk size.
+	 *
+	 * @return array
+	 */
+	public function get_next_chunk( $config, $status, $chunk_size ) {
+
+		$comment_ids = parent::get_next_chunk( $config, $status, $chunk_size );
+		if ( empty( $comment_ids ) ) {
+			return array();
+		}
+		$comments = get_comments(
+			array(
+				'include_unapproved' => true,
+				'comment__in'        => $comment_ids,
+				'orderby'            => 'comment_ID',
+				'order'              => 'DESC',
+			)
+		);
+		if ( empty( $comments ) ) {
+			return array();
+		}
+		return array(
+			'object_ids' => $comment_ids,
```

Comment:
What happens when we have inconsistent `$comment_ids`and `$comments`? eg `get_comments` returning less comments than the `$comment_ids` provided via `comment__in`

Reply:
So `object_ids` is used down the road to set the full sync `status` which needs to be aligned with the chunking logic. It is not sent to wpcom though, so inconsistencies are expected to happen

Reply:
As discussed internally, it makes sense to only fetch the metadata associated to $comments and not to all $comment_ids though. https://github.com/Automattic/jetpack/pull/41183/commits/ad8a0b6142face08a681fa9b4e205a91a4ed3ddf takes care of that